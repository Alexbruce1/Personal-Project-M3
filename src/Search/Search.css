.search {
  background-color: rgba(252, 252, 252, 1);
  border: none;
  box-shadow: 0 0 1rem rgba(0, 0, 0, .4);
  border-radius: 8px;
  height: 15rem;
  max-width: 45rem;
  margin: auto;
  min-width: 32rem;
  padding: 3rem 0 5rem;
  width: 50%;
}

.search-title {
  color: rgba(0, 0, 0, .7);
  font-family: 'Lato', sans-serif;
  font-style: italic;
  font-weight: 900;
  letter-spacing: 1px;
  margin: 0;
  padding: 0;
}

.search-input-container {
  background-color: #efefef;
  border: none;
  border-radius: 1rem;
  display: inline-flex;
  justify-content: space-between;
  padding: 12px 0;
  margin: 1.25rem 0;
  outline: none;
  transition-duration: .75s;
  width: 75%;
}

.search-input {
  background-color: transparent;
  border: none;
  text-indent: 1rem;
  color: #444;
  font-family: 'Lato', sans-serif;
  font-size: 1.25rem;
  font-weight: 600;
  letter-spacing: 1px;
  outline: none;
  overflow: hidden;
  width: 60%;
}

.search-input-container:focus-within {
  background-color: #fcfcfc;
  box-shadow: inset 0 0 8px rgba(0, 0, 0, .1);
}

.input-divider {
  background: #00000060;
  height: 2rem;
  width: 1px;
}

.type-select {
  background-color: transparent;
  border: none;
  color: #444;
  cursor: pointer;
  font-family: 'Lato', sans-serif;
  font-size: 1.25rem;
  font-weight: 200;
  letter-spacing: 1px;
  margin-right: 1rem;
  outline: none;
  width: 35%;
}

.search-submit,
.search-browse {
  border: none;
  border-radius: 1rem;
  font-family: 'Lato', sans-serif;
  font-size: 1.25rem;
  font-weight: 600;
  letter-spacing: 1px;
  padding: 12px 0;
  margin: 1.25rem 0;
  outline: none;
  text-indent: 1rem;
  transition-duration: .3s;
  width: 75%;
}

.search-submit:hover {
  box-shadow: 0 0 5px rgba(0, 0, 0, .7);
}

.search-submit {
  background-color: #D42F27;
  box-shadow: 0 1px 7px rgba(0, 0, 0, .3);
  color: #fcfcfc;
  cursor: pointer;
}

.search-browse {
  background-color: #D42F27;
  box-shadow: 0 1px 7px rgba(0, 0, 0, .3);
  color: #fcfcfc;
  cursor: pointer;
}

.search-browse:hover {
  box-shadow: 0 0 5px rgba(0, 0, 0, .9);
}