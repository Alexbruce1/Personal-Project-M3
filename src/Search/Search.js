import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Search.css';

class Search extends Component {
  
  getData = async (e) => {
    e.preventDefault();
    this.props.callApi();
  }

  render() {
    return(
      <div className='search'>
        <h2 className='search-title'>Search for content below</h2>
        <form className='search-form' onSubmit={this.getData}>
          <div className='search-input-container'>
            <input className='search-input' placeholder='Search' onChange={this.props.storeQuery}/>
            <div className='input-divider'></div>
            <select className='type-select' onChange={this.props.setType}>
              <option className='select-option' value='Everything'>Everything</option>
              <option className='select-option' value='Movies'>Movies</option>
              <option className='select-option' value='TV'>TV</option>
              <option className='select-option' value='Books'>Books</option>
              <option className='select-option' value='Authors'>Authors</option>
              <option className='select-option' value='Games'>Games</option>
              <option className='select-option' value='Music'>Music</option>
            </select>
          </div>
          <input className='search-submit' type='submit' value='Search' />
          {/* <input className='search-browse' type='button' value='Browse' /> */}
        </form>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  media: state.media
})

// export const mapDispatchToProps = (dispatch) => ({
//   newFunction: (data) => dispatch(getData(data))
// })


export default connect(mapStateToProps, null)(Search);