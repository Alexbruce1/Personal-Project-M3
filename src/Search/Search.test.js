import { Search, mapStateToProps } from './Search';
import React from 'react';
import { shallow } from 'enzyme';

describe('Search', () => {
  let wrapper;
  let callApi;
  let storeQuery;
  let setType;

  beforeEach(() => {
    callApi = jest.fn();
    storeQuery = jest.fn();
    setType = jest.fn();
    wrapper = shallow(<Search 
      callApi={callApi}
      storeQuery={storeQuery}
      setType={setType}
    />);
  });

  it('should render like the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show media', () => {
    const initialState = {
      media: []
    };

    expect(mapStateToProps(initialState).media).toEqual([]);
  });

  it('should call storeQuery on change', () => {
    wrapper.find('.search-input').simulate('change');

    expect(storeQuery).toHaveBeenCalled();
  });

  it('should call setType on change', () => {
    wrapper.find('.type-select').simulate('change');

    expect(setType).toHaveBeenCalled();
  });

  it('should call callApi on change', async() => {
    let e = {preventDefault: () => {}}
    await wrapper.instance().getData(e);

    await expect(callApi).toHaveBeenCalled();
  });
})