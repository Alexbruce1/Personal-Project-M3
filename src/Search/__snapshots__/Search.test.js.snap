// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render like the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    callApi={[MockFunction]}
    setType={[MockFunction]}
    storeQuery={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="search-title"
        >
          Search for content below
        </h2>,
        <form
          className="search-form"
          onSubmit={[Function]}
        >
          <div
            className="search-input-container"
          >
            <input
              className="search-input"
              onChange={[MockFunction]}
              placeholder="Search"
            />
            <div
              className="input-divider"
            />
            <select
              className="type-select"
              onChange={[MockFunction]}
            >
              <option
                className="select-option"
                value="Everything"
              >
                Everything
              </option>
              <option
                className="select-option"
                value="Movies"
              >
                Movies
              </option>
              <option
                className="select-option"
                value="TV"
              >
                TV
              </option>
              <option
                className="select-option"
                value="Books"
              >
                Books
              </option>
              <option
                className="select-option"
                value="Authors"
              >
                Authors
              </option>
              <option
                className="select-option"
                value="Games"
              >
                Games
              </option>
              <option
                className="select-option"
                value="Music"
              >
                Music
              </option>
            </select>
          </div>
          <input
            className="search-submit"
            type="submit"
            value="Search"
          />
        </form>,
      ],
      "className": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search for content below",
          "className": "search-title",
        },
        "ref": null,
        "rendered": "Search for content below",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="search-input-container"
            >
              <input
                className="search-input"
                onChange={[MockFunction]}
                placeholder="Search"
              />
              <div
                className="input-divider"
              />
              <select
                className="type-select"
                onChange={[MockFunction]}
              >
                <option
                  className="select-option"
                  value="Everything"
                >
                  Everything
                </option>
                <option
                  className="select-option"
                  value="Movies"
                >
                  Movies
                </option>
                <option
                  className="select-option"
                  value="TV"
                >
                  TV
                </option>
                <option
                  className="select-option"
                  value="Books"
                >
                  Books
                </option>
                <option
                  className="select-option"
                  value="Authors"
                >
                  Authors
                </option>
                <option
                  className="select-option"
                  value="Games"
                >
                  Games
                </option>
                <option
                  className="select-option"
                  value="Music"
                >
                  Music
                </option>
              </select>
            </div>,
            <input
              className="search-submit"
              type="submit"
              value="Search"
            />,
          ],
          "className": "search-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="search-input"
                  onChange={[MockFunction]}
                  placeholder="Search"
                />,
                <div
                  className="input-divider"
                />,
                <select
                  className="type-select"
                  onChange={[MockFunction]}
                >
                  <option
                    className="select-option"
                    value="Everything"
                  >
                    Everything
                  </option>
                  <option
                    className="select-option"
                    value="Movies"
                  >
                    Movies
                  </option>
                  <option
                    className="select-option"
                    value="TV"
                  >
                    TV
                  </option>
                  <option
                    className="select-option"
                    value="Books"
                  >
                    Books
                  </option>
                  <option
                    className="select-option"
                    value="Authors"
                  >
                    Authors
                  </option>
                  <option
                    className="select-option"
                    value="Games"
                  >
                    Games
                  </option>
                  <option
                    className="select-option"
                    value="Music"
                  >
                    Music
                  </option>
                </select>,
              ],
              "className": "search-input-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "search-input",
                  "onChange": [MockFunction],
                  "placeholder": "Search",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-divider",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      className="select-option"
                      value="Everything"
                    >
                      Everything
                    </option>,
                    <option
                      className="select-option"
                      value="Movies"
                    >
                      Movies
                    </option>,
                    <option
                      className="select-option"
                      value="TV"
                    >
                      TV
                    </option>,
                    <option
                      className="select-option"
                      value="Books"
                    >
                      Books
                    </option>,
                    <option
                      className="select-option"
                      value="Authors"
                    >
                      Authors
                    </option>,
                    <option
                      className="select-option"
                      value="Games"
                    >
                      Games
                    </option>,
                    <option
                      className="select-option"
                      value="Music"
                    >
                      Music
                    </option>,
                  ],
                  "className": "type-select",
                  "onChange": [MockFunction],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Everything",
                      "className": "select-option",
                      "value": "Everything",
                    },
                    "ref": null,
                    "rendered": "Everything",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Movies",
                      "className": "select-option",
                      "value": "Movies",
                    },
                    "ref": null,
                    "rendered": "Movies",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "TV",
                      "className": "select-option",
                      "value": "TV",
                    },
                    "ref": null,
                    "rendered": "TV",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Books",
                      "className": "select-option",
                      "value": "Books",
                    },
                    "ref": null,
                    "rendered": "Books",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Authors",
                      "className": "select-option",
                      "value": "Authors",
                    },
                    "ref": null,
                    "rendered": "Authors",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Games",
                      "className": "select-option",
                      "value": "Games",
                    },
                    "ref": null,
                    "rendered": "Games",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Music",
                      "className": "select-option",
                      "value": "Music",
                    },
                    "ref": null,
                    "rendered": "Music",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-submit",
              "type": "submit",
              "value": "Search",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="search-title"
          >
            Search for content below
          </h2>,
          <form
            className="search-form"
            onSubmit={[Function]}
          >
            <div
              className="search-input-container"
            >
              <input
                className="search-input"
                onChange={[MockFunction]}
                placeholder="Search"
              />
              <div
                className="input-divider"
              />
              <select
                className="type-select"
                onChange={[MockFunction]}
              >
                <option
                  className="select-option"
                  value="Everything"
                >
                  Everything
                </option>
                <option
                  className="select-option"
                  value="Movies"
                >
                  Movies
                </option>
                <option
                  className="select-option"
                  value="TV"
                >
                  TV
                </option>
                <option
                  className="select-option"
                  value="Books"
                >
                  Books
                </option>
                <option
                  className="select-option"
                  value="Authors"
                >
                  Authors
                </option>
                <option
                  className="select-option"
                  value="Games"
                >
                  Games
                </option>
                <option
                  className="select-option"
                  value="Music"
                >
                  Music
                </option>
              </select>
            </div>
            <input
              className="search-submit"
              type="submit"
              value="Search"
            />
          </form>,
        ],
        "className": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for content below",
            "className": "search-title",
          },
          "ref": null,
          "rendered": "Search for content below",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="search-input-container"
              >
                <input
                  className="search-input"
                  onChange={[MockFunction]}
                  placeholder="Search"
                />
                <div
                  className="input-divider"
                />
                <select
                  className="type-select"
                  onChange={[MockFunction]}
                >
                  <option
                    className="select-option"
                    value="Everything"
                  >
                    Everything
                  </option>
                  <option
                    className="select-option"
                    value="Movies"
                  >
                    Movies
                  </option>
                  <option
                    className="select-option"
                    value="TV"
                  >
                    TV
                  </option>
                  <option
                    className="select-option"
                    value="Books"
                  >
                    Books
                  </option>
                  <option
                    className="select-option"
                    value="Authors"
                  >
                    Authors
                  </option>
                  <option
                    className="select-option"
                    value="Games"
                  >
                    Games
                  </option>
                  <option
                    className="select-option"
                    value="Music"
                  >
                    Music
                  </option>
                </select>
              </div>,
              <input
                className="search-submit"
                type="submit"
                value="Search"
              />,
            ],
            "className": "search-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="search-input"
                    onChange={[MockFunction]}
                    placeholder="Search"
                  />,
                  <div
                    className="input-divider"
                  />,
                  <select
                    className="type-select"
                    onChange={[MockFunction]}
                  >
                    <option
                      className="select-option"
                      value="Everything"
                    >
                      Everything
                    </option>
                    <option
                      className="select-option"
                      value="Movies"
                    >
                      Movies
                    </option>
                    <option
                      className="select-option"
                      value="TV"
                    >
                      TV
                    </option>
                    <option
                      className="select-option"
                      value="Books"
                    >
                      Books
                    </option>
                    <option
                      className="select-option"
                      value="Authors"
                    >
                      Authors
                    </option>
                    <option
                      className="select-option"
                      value="Games"
                    >
                      Games
                    </option>
                    <option
                      className="select-option"
                      value="Music"
                    >
                      Music
                    </option>
                  </select>,
                ],
                "className": "search-input-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "search-input",
                    "onChange": [MockFunction],
                    "placeholder": "Search",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input-divider",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        className="select-option"
                        value="Everything"
                      >
                        Everything
                      </option>,
                      <option
                        className="select-option"
                        value="Movies"
                      >
                        Movies
                      </option>,
                      <option
                        className="select-option"
                        value="TV"
                      >
                        TV
                      </option>,
                      <option
                        className="select-option"
                        value="Books"
                      >
                        Books
                      </option>,
                      <option
                        className="select-option"
                        value="Authors"
                      >
                        Authors
                      </option>,
                      <option
                        className="select-option"
                        value="Games"
                      >
                        Games
                      </option>,
                      <option
                        className="select-option"
                        value="Music"
                      >
                        Music
                      </option>,
                    ],
                    "className": "type-select",
                    "onChange": [MockFunction],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Everything",
                        "className": "select-option",
                        "value": "Everything",
                      },
                      "ref": null,
                      "rendered": "Everything",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Movies",
                        "className": "select-option",
                        "value": "Movies",
                      },
                      "ref": null,
                      "rendered": "Movies",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "TV",
                        "className": "select-option",
                        "value": "TV",
                      },
                      "ref": null,
                      "rendered": "TV",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Books",
                        "className": "select-option",
                        "value": "Books",
                      },
                      "ref": null,
                      "rendered": "Books",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Authors",
                        "className": "select-option",
                        "value": "Authors",
                      },
                      "ref": null,
                      "rendered": "Authors",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Games",
                        "className": "select-option",
                        "value": "Games",
                      },
                      "ref": null,
                      "rendered": "Games",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Music",
                        "className": "select-option",
                        "value": "Music",
                      },
                      "ref": null,
                      "rendered": "Music",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search-submit",
                "type": "submit",
                "value": "Search",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
